marc_export/src/main/java/whelk/export/marc/ProfileExport.java

```diff
@@ -430,4 +460,206 @@ private PreparedStatement getAllChangedByStatement(String id, Timestamp from, Ti
         preparedStatement.setString(3, id);
         return preparedStatement;
     }
+
+    /**
+     * Prepare and convert documents to MARC in parallel.
+     *
+     * Work is performed by a shared pool handling all requests. (Using more threads will just slow us down.) 
+     * Serializing and writing the converted records to the output stream is done by a separate single thread per request. 
+     * 
+     * Getting ids to be exported from the database will always be faster than exporting them.
+     * So we need to block the calling thread when too much work is queued up.
+     * Alas, there is no clean way of having a ThreadPoolExecutor with a bounded queue where the caller blocks
```

Comment:
I don't agree it's simpler. And since this thread is waiting >99% of the time it would mostly be scheduled to run without anything to do.

ThreadPool isn't really a thread pool as it doesn't reuse threads but starts a new one for each task. Maybe not a big problem here since tasks are relatively long running. 

I'd rather replace it instead.